### ISSUE TITLE
GraphQL & Gatsby 3, 4 - not working

### ISSUE THREAD

**Issue by MikeKry** [OrchardCMS/OrchardCore#10781]
OrchardCore does not work with new gatsby versions. 

It seems that it is caused by different data type structures. I have tested update of GraphQL .NET update and it helped. 

I have also created a PR that resolves this issue.

stacktrace from error here:

Missing onError handler for invocation 'building-schema', error was 'RangeError: Maximum call stack size exceeded'. Stacktrace was 'RangeError: Maximum call stack size exceeded
at getInputObjectTypeDefinitionNode (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\utils\definitionNode.js:49:3)
at InputTypeComposer.getType (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\InputTypeComposer.js:436:81)
at ListComposer.getType (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\ListComposer.js:22:49)
at InputTypeComposer.getFieldType (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\InputTypeComposer.js:315:42)
at D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\utils\definitionNode.js:349:101
at Array.map (<anonymous>)
at getInputValueDefinitionNodes (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\utils\definitionNode.js:336:21)
at getInputObjectTypeDefinitionNode (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\utils\definitionNode.js:57:13)
at InputTypeComposer.getType (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\InputTypeComposer.js:436:81)
at ListComposer.getType (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\ListComposer.js:22:49)
at InputTypeComposer.getFieldType (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\InputTypeComposer.js:315:42)
at D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\utils\definitionNode.js:349:101
at Array.map (<anonymous>)
at getInputValueDefinitionNodes (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\utils\definitionNode.js:336:21)
at getInputObjectTypeDefinitionNode (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\utils\definitionNode.js:57:13)
at InputTypeComposer.getType (D:\project\project.Gatsby\node_modules\gatsby\node_modules\graphql-compose\lib\InputTypeComposer.js:436:81)'

**Comment by MikeKry** [comment]
adding ref. to open PR #9087

**Comment by Piedone** [comment]
We've updated GraphQL many times since this. Can you tell if this is still an issue?

**Comment by hishamco** [comment]
Ya, I will never remember the massive change that I started on GraphQL, then completed in an elegant way by @hyzx86. I think this should be fixed in the latest bits

**Comment by MikeKry** [comment]
@Piedone 

No, sorry I forgot to close this one :) 

This is fixed by https://github.com/OrchardCMS/OrchardCore/pull/9087 that we had done long time ago, but it took kinda long to merge so I forgot because I was locally already using version containing my last commit.

That said - it is working in OrchardCore 1.4 to 1.8. But as @hishamco mentioned, there was another update of GraphQL done (https://github.com/OrchardCMS/OrchardCore/pull/15284/files) and I noticed that there were multiple bug reports. That is reason why I did not switch to latest version of OrchardCore yet and I do not know if it works on new version.

But I think we can close it for now and when I will eventually update locally, then I can reopen if this problem occurs again?
